firmware/quadruna/CRIT/src/app/app_mainState.c

```diff
@@ -22,20 +23,24 @@ static void mainStateRunOnTick100Hz(void)
     const bool ams_fault_latched = app_canRx_BMS_BmsLatchedFault_get();
     io_led_enable(globals->config->ams_led, ams_fault_latched);
 
-    const bool shutdown_sensor = hw_gpio_readPin(globals->config->shdn_sen_pin);
+    const bool shutdown_sensor = io_shutdownSensor_readPin(globals->config->shdn_sen);
     io_led_enable(globals->config->shdn_led, shutdown_sensor);
 
     const bool start_switch_on = io_switch_isClosed(globals->config->start_switch);
     app_canTx_CRIT_StartSwitch_set(start_switch_on ? SWITCH_ON : SWITCH_OFF);
-    io_led_enable(globals->config->start_led, start_switch_on);
 
     const bool regen_switch_on = io_switch_isClosed(globals->config->regen_switch);
     app_canTx_CRIT_RegenSwitch_set(regen_switch_on ? SWITCH_ON : SWITCH_OFF);
-    io_led_enable(globals->config->regen_led, regen_switch_on);
 
     const bool torquevec_switch_on = io_switch_isClosed(globals->config->torquevec_switch);
     app_canTx_CRIT_TorqueVecSwitch_set(torquevec_switch_on ? SWITCH_ON : SWITCH_OFF);
-    io_led_enable(globals->config->torquevec_led, torquevec_switch_on);
+
+    if (app_canRx_VC_State_get() == 2)
+    {
+        io_led_enable(globals->config->start_led, true);
+        io_led_enable(globals->config->regen_led, true);
+        io_led_enable(globals->config->torquevec_led, true);
```

Comment:
idk about the logic here, i don't think this is correct. state drives the start led but not the regen/torque vectoring LEDs.

Reply:
ah ok i see. ok i moved the regen/torque vec LEDs to being read from switches.